src/workerd/api/node/crypto_random.c++

```diff
@@ -0,0 +1,85 @@
+#include "crypto.h"
+#include "crypto_util.h"
+#include <v8.h>
+#include <openssl/bn.h>
+#include <openssl/rand.h>
+#include <workerd/jsg/jsg.h>
+#include <workerd/api/crypto-impl.h>
+
+namespace workerd::api::node {
+  kj::Array<kj::byte> CryptoImpl::randomPrime(uint32_t size, bool safe,
+      jsg::Optional<kj::Array<kj::byte>> add_buf, jsg::Optional<kj::Array<kj::byte>> rem_buf) {
+  BIGNUM* add = nullptr;
+
+  // Use mapping to have kj::Own work with optional buffer, perhaps there's a better way to format this
+  auto _add = add_buf.map([](kj::Array<kj::byte>& a) {
+      return OSSLCALL_OWN(BIGNUM, BN_bin2bn(a.begin(), a.size(), nullptr),
+      InternalDOMOperationError, "Error importing add parameter",
+      internalDescribeOpensslErrors());
+  });
+  KJ_IF_MAYBE(a, _add) {
+      add = a->get();
+  }
+
+  BIGNUM* rem = nullptr;
+  auto _rem = rem_buf.map([](kj::Array<kj::byte>& r) {
+      return OSSLCALL_OWN(BIGNUM, BN_bin2bn(r.begin(), r.size(), nullptr),
+      InternalDOMOperationError, "Error importing rem parameter",
+      internalDescribeOpensslErrors());
+  });
+  KJ_IF_MAYBE(r, _rem) {
+      rem = r->get();
+  }
+
+  // The JS interface already ensures that the (positive) size fits into an int.
+  int bits = static_cast<int>(size);
+
+  if (add) {
+      // If we allowed this, the best case would be returning a static prime
+      // that wasn't generated randomly. The worst case would be an infinite
+      // loop within OpenSSL, blocking the main thread or one of the threads
+      // in the thread pool.
+    JSG_REQUIRE(BN_num_bits(add) <= bits, RangeError, "invalid options.add");
```

Comment:
I'd prefer this error message and the one immediately below to be a bit more descriptive, letting the user know why the values are invalid.